controller/controller.js

```diff
@@ -12,5 +12,78 @@ exports.getApiTopics = (request, response) => {
     });
 };
 
+// /api/articles Endpoints
+exports.getApiArticles = (request, response) => {
+    // TODO: instead of doing this for every articles request, find way to just store comment count in the db and keep it updated, that way theres no need to calculate it each time
+    model.selectArticles().then((selectArticlesQueryResult) => {
+        // queryResult = selected articles
+        // TODO: move this stuff from controller to model, data manipulation is meant to be in the model only
+        const articlesArray = selectArticlesQueryResult.rows;
+        
+        // TODO: consider changing below to use MAP instead of foreach?
+        const buildArticleIdArray = function (articleElement) {
+            articleIdArray.push(articleElement["article_id"]);
+        };
+
+        // for each article ID in articlesArray, store ID in its own array called articleIdArray
+        const articleIdArray = [];
+        articlesArray.forEach(buildArticleIdArray);
+
```

Comment:
You can use a simple SQL query to do all this, this shouldn't be performed with js functions or more than one call to the db